OpenRA.Mods.Common/Traits/Render/SupportPowerChargeBar.cs

```diff
@@ -15,38 +15,51 @@
 
 namespace OpenRA.Mods.Common.Traits.Render
 {
-	[Desc("Display the time remaining until the super weapon attached to the actor is ready to the player and his allies.")]
+	[Desc("Display the time remaining until the super weapon attached to the actor is ready to the player, his allies",
+		"and if 'DisplayTimer' on the support power is true, to all players specified through the 'DisplayTimerStances' field.")]
 	class SupportPowerChargeBarInfo : ITraitInfo
 	{
 		public readonly Color Color = Color.Magenta;
 
 		public object Create(ActorInitializer init) { return new SupportPowerChargeBar(init.Self, this); }
 	}
 
-	class SupportPowerChargeBar : ISelectionBar
+	class SupportPowerChargeBar : ISelectionBar, INotifyOwnerChanged
 	{
 		readonly Actor self;
 		readonly SupportPowerChargeBarInfo info;
+		SupportPowerManager spm;
 
 		public SupportPowerChargeBar(Actor self, SupportPowerChargeBarInfo info)
 		{
 			this.self = self;
 			this.info = info;
+			spm = self.Owner.PlayerActor.Trait<SupportPowerManager>();
 		}
 
 		float ISelectionBar.GetValue()
 		{
-			if (!self.Owner.IsAlliedWith(self.World.RenderPlayer))
-				return 0;
-
-			var spm = self.Owner.PlayerActor.Trait<SupportPowerManager>();
 			var power = spm.GetPowersForActor(self).FirstOrDefault(sp => !sp.Disabled);
+			if (power == null)
+				return 0;
 
-			if (power == null) return 0;
+			if (power.Info.DisplayTimer)
+			{
+				var viewer = self.World.RenderPlayer ?? self.World.LocalPlayer;
+				if (viewer != null && !power.Info.DisplayTimerStances.HasStance(self.Owner.Stances[viewer]))
```

Comment:
There are cases where mods might want to show the bar but not the timer, so it would be better to have a separate `DisplayBarStances` field for this.